tests/mpi/step-37.cc

```diff
@@ -0,0 +1,568 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2018 by the deal.II authors
+ *
+ * This file is part of the deal.II library.
+ *
+ * The deal.II library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the deal.II distribution.
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * a light modification of step-37 tutorial to test MappingFEField
+ * in parallel with non-zero BC and zero body load.
```

Comment:
This is a pretty large test for such a small function. Can we also have a test that shows wrong results without this PR?

Reply:
@masterleinad done.